src/Jackalope/Property.php

```diff
@@ -709,25 +685,44 @@ public function _setValue($value, $type = PropertyType::UNDEFINED)
          */
 
         if (PropertyType::UNDEFINED === $type) {
-            $type = $this->valueConverter->determineType(is_array($value) ? reset($value) : $value);
+            if (is_array($value) && 0 === count($value)) {
+                $type = PropertyType::STRING;
```

Comment:
This repeats the logic from before .. can it / would it make sense for it to be moved into the value converter?